tests/test_distributions.py

```diff
@@ -111,3 +112,23 @@ def test_categorical(dist, CAT_ACTIONS):
     entropy = dist.entropy()
     log_prob = dist.log_prob(actions)
     assert th.allclose(entropy.mean(), -log_prob.mean(), rtol=5e-3)
+
+
+@pytest.mark.parametrize(
+    "dist_type",
+    [
+        BernoulliDistribution(N_ACTIONS).proba_distribution(th.rand(N_ACTIONS)),
+        CategoricalDistribution(N_ACTIONS).proba_distribution(th.rand(N_ACTIONS)),
+        DiagGaussianDistribution(N_ACTIONS).proba_distribution(th.rand(N_ACTIONS), th.rand(N_ACTIONS)),
+        MultiCategoricalDistribution([N_ACTIONS, N_ACTIONS]).proba_distribution(th.rand(1, sum([N_ACTIONS, N_ACTIONS]))),
+        SquashedDiagGaussianDistribution(N_ACTIONS).proba_distribution(th.rand(N_ACTIONS), th.rand(N_ACTIONS)),
+        StateDependentNoiseDistribution(N_ACTIONS).proba_distribution(
+            th.rand(N_ACTIONS), th.rand([N_ACTIONS, N_ACTIONS]), th.rand([N_ACTIONS, N_ACTIONS])
+        ),
+    ],
+)
+def test_kl_divergence(dist_type):
+    dist1 = dist_type
+    dist2 = dist_type
+
+    assert kl_divergence(dist1, dist2) == 0
```

Comment:
I would also add a check that compare the kl div here with the one you can compute by sampling (see what is done for the entropy)

Reply:
I would also do some sanity checks when we know in advance the value of the kl div (for which the kl is not zero, for instance comparing two 1D Gaussian dist)

Reply:
sure I can add these :)